core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageAgentResourceConfigStep.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.deployment.steps;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.builder.JsonReader;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourcePatternsBuildItem;
+import io.quarkus.deployment.pkg.builditem.BuildSystemTargetBuildItem;
+import io.quarkus.deployment.pkg.steps.NativeOrNativeSourcesBuild;
+
+public class NativeImageAgentResourceConfigStep {
+    private static final Logger log = Logger.getLogger(NativeImageAgentResourceConfigStep.class);
+
+    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
+    void transformResourceConfig(BuildProducer<NativeImageResourcePatternsBuildItem> resourcePatternProducer,
+            BuildSystemTargetBuildItem buildSystemTargetBuildItem) {
+        final Path agentResourceConfigPath = buildSystemTargetBuildItem.getOutputDirectory()
+                .resolve(Path.of("generated-native-config", "resource-config.json"));
+        if (agentResourceConfigPath.toFile().exists()) {
+            final Pattern ignoreResources = Pattern.compile(
+                    "^((?!(microprofile|quarkus|smallrye|jboss|junit|slf4j|application.properties|surefire|logging.properties|Test.class|java/lang|java/util)).)*$");
+            try {
+                final NativeImageResourcePatternsBuildItem.Builder patternBuilder = NativeImageResourcePatternsBuildItem
+                        .builder();
+                final String agentResourceConfig = Files.readString(agentResourceConfigPath);
+                final JsonReader jsonReader = JsonReader.of(agentResourceConfig);
+                final JsonReader.JsonObject resources = jsonReader.<JsonReader.JsonObject> read().get("resources");
+                final JsonReader.JsonArray includes = resources.get("includes");
+                final Stream<JsonReader.JsonObject> stream = includes.stream();
+                stream.<JsonReader.JsonString> map(include -> include.get("pattern"))
+                        .map(JsonReader.JsonString::value)
+                        .filter(pattern -> ignoreResources.matcher(pattern).find())
+                        .forEach(patternBuilder::includeGlob);
```

Comment:
@galderz I think the following resolves https://github.com/quarkusio/quarkus/issues/36823

```suggestion
                        .forEach(patternBuilder::includePattern);
```

Reply:
Ah ok. I'm going with a diffeerent approach where I won't be using this anyway. I'll close #36823.